Pegasus/include/geometry.hpp

```diff
@@ -0,0 +1,880 @@
+#ifndef PEGASUS_GEOMETRY_HPP
+#define PEGASUS_GEOMETRY_HPP
+
+#include "Pegasus/include/math.hpp"
+
+#include <algorithm>
+#include <array>
+#include <functional>
+#include <limits>
+#include <memory>
+#include <cmath>
+#include <unordered_map>
+#include <vector>
+#include <utility>
+#include <cstdint>
+
+namespace pegasus {
+namespace geometry {
+
+    //Shapes
+    class Shape {
+    public:
+        explicit Shape(Vector3 const & centerOfMass);
+
+        void setCenterOfMass(Vector3 const & centerOfMass);
+        Vector3 getCenterOfMass() const;
```

Comment:
Would it make sense to return a `const&` instead of a copy? The client code itself may decide if it wants a copy or not. Same goes to other getters of non-basic types.